Source/Diagnostics/ComputeDiagFunctors/ParticleReductionFunctor.cpp

```diff
@@ -93,9 +93,8 @@ ParticleReductionFunctor::operator() (amrex::MultiFab& mf_dst, const int dcomp,
                 const amrex::ParticleReal ux = p.rdata(PIdx::ux) / PhysConst::c;
                 const amrex::ParticleReal uy = p.rdata(PIdx::uy) / PhysConst::c;
                 const amrex::ParticleReal uz = p.rdata(PIdx::uz) / PhysConst::c;
-                amrex::Real value;
-                if ((do_filter) && (!filter_fn(xw, yw, zw, ux, uy, uz))) value = 0._rt;
-                else value = map_fn(xw, yw, zw, ux, uy, uz);
+                const bool filtered_out_flag = ((do_filter) && (filter_fn(xw, yw, zw, ux, uy, uz) <= 0.0));
```

Comment:
```suggestion
                const bool filtered_out_flag = ((do_filter) && (filter_fn(xw, yw, zw, ux, uy, uz) == 0.0_prt));
```

Reply:
I am not totally sure about this. `filter_fn` is  an `amrex::ParserExecutor`. Doesn't the `()` operator return a `double` ?

Reply:
The return type is the same as the arguments (ParticleReal in this case).

Reply:
Ah, I see.. I stopped at the first definition. I didn't see the one with `std::enable_if_t<sizeof...(Ts) == N &&  amrex::Same<float,Ts...>::value, float>` . You are totally right